pkg/process/checks/process.go

```diff
@@ -243,11 +263,24 @@ func (p *ProcessCheck) runWLM(groupID int32, collectRealTime bool) (RunResult, e
 		return nil, errEmptyCPUTime
 	}
 
-	procs, err := p.probe.ProcessesByPID(time.Now(), true)
+	wlmProcList := p.wmeta.ListProcesses()
+	pids := make([]int32, len(wlmProcList))
+	for i, wlmProc := range wlmProcList {
+		pids[i] = wlmProc.Pid
+	}
+
+	statsForProcess, err := p.probe.StatsForPIDs(pids, p.timer.Now())
 	if err != nil {
 		return nil, err
 	}
 
+	// maintaining pre-existing variable types for ease of testing (don't have to change much following code)
```

Comment:
```suggestion
	// map to common process type used by other versions of the check
```
I think the rest is unnecessary.